google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -102,25 +104,67 @@ class BigtableStackdriverExportUtils {
   private static final LabelKey CLIENT_UID_LABEL_KEY =
       LabelKey.create(BuiltinMeasureConstants.CLIENT_UID.getName(), "client uid");
 
+  static boolean isBigtableTableMetric(MetricDescriptor metricDescriptor) {
+    return !metricDescriptor.getName().equals(PER_CONNECTION_ERROR_COUNT_VIEW.getName().asString());
+  }
+
+  static boolean shouldExportMetric(MetricDescriptor metricDescriptor) {
+    return metricDescriptor.getName().contains("bigtable")
+        && (isBigtableTableMetric(metricDescriptor)
+            || ConsumerEnvironmentUtils.isEnvGce()
+            || ConsumerEnvironmentUtils.isEnvGke());
+  }
+
   static com.google.monitoring.v3.TimeSeries convertTimeSeries(
       MetricDescriptor metricDescriptor,
       TimeSeries timeSeries,
       String clientId,
-      MonitoredResource monitoredResource) {
-    String metricName = metricDescriptor.getName();
-    List<LabelKey> labelKeys = metricDescriptor.getLabelKeys();
+      MonitoredResource bigtableMonitoredResource,
+      MonitoredResource gceMonitoredResource,
+      MonitoredResource gkeMonitoredResource) {
     Type metricType = metricDescriptor.getType();
 
-    MonitoredResource.Builder monitoredResourceBuilder = monitoredResource.toBuilder();
+    com.google.monitoring.v3.TimeSeries.Builder builder;
+    if (isBigtableTableMetric(metricDescriptor)) {
+      builder =
+          createBuilderForBigtableResource(
```

Comment:
let's rename these methods to "setupBuilderForBigtableResource" since we're also modifying the builder.